### ISSUE TITLE
symfony form and contao core

### ISSUE THREAD

**Issue by siggj** [contao/core#8080]
I injected the contents of a template twig in a Contao's article. Everything is going well.
But, when I inject a content with a form in Contao, Contao controls the token because he thinks it's a Contao form?
Can I avoid this?

**Comment by baumannsven** [comment]
@siggj 
See you that report. It works.

https://github.com/ContaoBlackForest/contao-symfony-component-form

**Comment by siggj** [comment]
Ok but how it works ?
And what this component make ? Do you have some documentation ?
Thanks in advance for your response

**Comment by baumannsven** [comment]
@siggj 
This https://github.com/ContaoBlackForest/contao-calendar-filter/tree/develop used

**Comment by siggj** [comment]
Well, thanks for the link.
But it's not exactly what I need. I really made a controller and an action, and for a specific route that represent the alias of contao page, i inject some content from this action.
I had to handle request and check is form is valid in the controller.

**Comment by leofeyer** [comment]
I've had the same problem today while refactoring the install tool. If it is a Contao route, you will have to include the request token in your forms and send it together with the other POST data.

@contao/developers We should probably make this configurable in the container?

**Comment by siggj** [comment]
OK thanks for this answer. But one more point, if I include the request token in the form, I have to use the class RequestToken from Contao. Right ? 
And if i have to use this, i have to initialize the framework. But in the initialization method, there is a call to the method "handleRequestToken"...

**Comment by leofeyer** [comment]
The framework should be initialized already, otherwise you would not have gotten the request token error in the first place.

**Comment by Toflar** [comment]
But you wouldn't really need it :) For the record:

```php
$token = $container->get('security.csrf.token_manager')->getToken($container->getParameter('contao.csrf_token_name'))->getValue();
```

Obviously you should inject the service and parameter but you get the point :)

**Comment by aschempp** [comment]
Also there was a constant to disable the request token previously. We should find an alternative to that, maybe a route attribute? Won't solve the problem for custom routes though…

**Comment by siggj** [comment]
@Toflar yes we can do with the token_manager and it works but I don't think it's a real solution because we don't have to pass absolutely a token. Well, If we use contao form, then contao must validate token. If we use symfony form, then symfony must validate...

@leofeyer thanks for the improvement, now we can do really what we want like this. So initialize the framework and set this boolean right after that.

What now if there are a form in the content injected by Symfony and a form in another place like right or left column from contao ? So one had to be validate by Symfony and the other one by Contao, right ?

**Comment by siggj** [comment]
In one action of a controller :
1. Prepare data we want to display
2. handle request if (symfony) form is submitted
3. render template we want to innject
4. Run contao frontend and let contao do his personnal job (display, validate contao form, ...)

So I think we want to disable token check from point 1 to point 3 included. Then we want to enable token check.

Just say me if i'm wrong...

**Comment by leofeyer** [comment]
@aschempp A route attribute is the best solution. And it will not change the situation for custom extensions, because even in Contao 3, you have to provide a custom entry point to bypass the check.

**Comment by leofeyer** [comment]
I just realize that we have introduced two BC breaks in our new implementation:

```php
/**
 * Validate a token
 *
 * @param string $strToken The request token
 *
 * @return boolean True if the token matches the stored one
 */
public static function validate($strToken)
{
	##
	# BC break 1: neither disableRefererCheck nor BYPASS_TOKEN_CHECK are checked anymore
	##
	// The feature has been disabled
	if (\Config::get('disableRefererCheck') || defined('BYPASS_TOKEN_CHECK'))
	{
		return true;
	}

	// Validate the token
	if ($strToken != '' && static::$strToken != '' && $strToken == static::$strToken)
	{
		return true;
	}

	##
	# BC break 2: the requestTokenWhitelist is not checked anymore
	##
	// Check against the whitelist (thanks to Tristan Lins) (see #3164)
	if (\Config::get('requestTokenWhitelist'))
	{
		$strHostname = gethostbyaddr($_SERVER['REMOTE_ADDR']);
		foreach (\Config::get('requestTokenWhitelist') as $strDomain)
		{
			if ($strDomain == $strHostname || preg_match('/\.' . preg_quote($strDomain, '/') . '$/', $strHostname))
			{
				return true;
			}
		}
	}
	return false;
}
```

What shall we do about this?

**Comment by leofeyer** [comment]
Forget about my last comment :)

**Comment by leofeyer** [comment]
And since we are no longer using the `RequestToken::validate()` method, we are neither checking for `disableRefererCheck` nor for `BYPASS_TOKEN_CHECK` nor for `requestTokenWhitelist`. This is a BC break, isn't it?

**Comment by aschempp** [comment]
I remember we discussed something about it, but I can't really remember…

**Comment by Toflar** [comment]
Yes, it must be a config param.

**Comment by leofeyer** [comment]
We have discussed this issue on Mumble.

1. You will never be able to use forms in Contao that have no CSRF token at all.

2. Right now, you have to add the Contao request token to make it work. However, we want to try to completely drop our request token check in favor of the Symfony CSRF check (in a backwards compatible way).

**Comment by leofeyer** [comment]
Fixed via contao/core-bundle#389.
